controllers/address.controller.js

```diff
@@ -0,0 +1,107 @@
+const User = require("../models/user.model");
+const Address = require("../models/adress.model");
+const { extend, concat, add } = require("lodash");
+
+const getAddresses = async (req, res) => {
+  const address = await Address.find({});
+  res.json({ success: true, address });
+};
+
+const findUserAddress = async (req, res, next, userId) => {
+  try {
+    const user = await User.findOne({ _id: userId });
+    if (!user) {
+      res.status(404).json({
+        success: false,
+        message: "Invalid user! Kindly register to continue",
+      });
+      throw Error("Invalid User");
+    }
```

Comment:
Hey I have one doubt here, why do you have to throw Error if you are sending response in line number 14, a return statement should work here.